itest/rust/src/builtin_tests/script/script_instance_tests.rs

```diff
@@ -222,3 +245,109 @@ impl ScriptInstance for TestScriptInstance {
         None
     }
 }
+
+#[derive(GodotClass)]
+#[class(base = ScriptLanguageExtension, tool, init)]
+struct TestScriptLanguage {
+    base: Base<ScriptLanguageExtension>,
+}
+
+#[godot_api]
+impl TestScriptLanguage {
+    #[func]
+    fn new_script(&self) -> Gd<TestScript> {
+        TestScript::new(self.base().to_godot().cast())
+    }
+}
+
+#[godot_api]
+#[rustfmt::skip]
+impl IScriptLanguageExtension for TestScriptLanguage {
```

Comment:
🙈 

You've used these APIs more than me; do you typically override many of the functions?

Do you think it's eventually worthwhile if the library provides a way to generate default implementations for the functions (although it may be hard to do generically...)

Reply:
I don't think there is a helpful default for most of these functions. In a real implementation, all these functions have to interact with the script engine to provide a good integration into the editor. We just get a way with the `unreachable!()` implementations in tests, since we are not testing the editor.